### ISSUE TITLE
error: deployment_eth_rpc_request_duration

### ISSUE THREAD

**Issue by kgcdream2019** [graphprotocol/graph-node#2750]
my local graph-node stuck from following error:

"registering metric [deployment_eth_rpc_request_duration] because it was already registered, component: MetricsRegistry"

Does anyone know the reason?

**Comment by leoyvens** [comment]
That error is benign, the real cause for your issue must be something else.

**Comment by cauta** [comment]
i met this issue too. Any reason?
```
Sep 03 08:53:26.039 ERRO registering metric [deployment_sync_secs] because it was already registered, component: MetricsRegistry
Sep 03 08:53:26.039 ERRO registering metric [deployment_eth_rpc_request_duration] because it was already registered, component: MetricsRegistry
Sep 03 08:53:26.039 ERRO registering metric [deployment_eth_rpc_errors] because it was already registered, component: MetricsRegistry
Sep 03 08:53:26.039 ERRO registering metric [deployment_block_trigger_count] because it was already registered, component: MetricsRegistry
Sep 03 08:53:26.039 ERRO registering metric [deployment_trigger_processing_duration] because it was already registered, component: MetricsRegistry
Sep 03 08:53:26.039 ERRO registering metric [deployment_block_processing_duration] because it was already registered, component: MetricsRegistry
Sep 03 08:53:26.039 ERRO registering metric [deployment_transact_block_operations_duration] because it was already registered, component: MetricsRegistry
Sep 03 08:53:26.039 ERRO registering metric [deployment_handler_execution_time] because it was already registered, component: MetricsRegistry
Sep 03 08:53:26.039 ERRO registering metric [deployment_host_fn_execution_time] because it was already registered, component: MetricsRegistry
Sep 03 08:53:26.039 ERRO registering metric [deployment_reverted_blocks] because it was already registered, component: MetricsRegistry
Sep 03 08:53:26.039 ERRO registering metric [deployment_head] because it was already registered, component: MetricsRegistry
Sep 03 08:53:26.039 ERRO registering metric [deployment_failed] because it was already registered, component: MetricsRegistry
Sep 03 08:53:26.060 ERRO registering metric [deployment_eth_rpc_request_duration] because it was already registered, component: MetricsRegistry
Sep 03 08:53:26.060 ERRO registering metric [deployment_eth_rpc_errors] because it was already registered, component: MetricsRegistry

```

**Comment by leoyvens** [comment]
Its a minor bug that this is happening but it shouldn't cause any real issues.

**Comment by TobyKreiselmaier** [comment]
Same for me
Apr 06 21:10:11.390 ERRO registering metric [deployment_eth_rpc_request_duration] because it was already registered, component: MetricsRegistry
Apr 06 21:10:11.390 ERRO registering metric [deployment_eth_rpc_errors] because it was already registered, component: MetricsRegistry
After that all RPC calls fail, that work on older graph nodes such as 0.19.2 and 0.22.0. Please @leoyvens what can be done about this?

**Comment by lutter** [comment]
This error won't make RPC calls fail - there must be something else going on. The error with metrics registration was fixed in PR #3426  but besides this error message the only effect of this was that some metrics were recorded incorrectly. Feel free to open another issue with more background on the failing RPC requests.

**Comment by air3ijai** [comment]
We also see such errors in the `graph-node:v0.26.0` logs
```
Jun 23 08:41:25.543 INFO Applying 3 entity operation(s), block_hash: 0x2a17c026a0aa65336a5f123634e5163b5b31bef3fc542250ef71fc9ded905d54, block_number: 10901391, sgd: 3, subgraph_id: QmdNChb53fQdvKvndqT3dbM2Hcoo5rPqU45iHmnnNtinf4, component: SubgraphInstanceManager
Jun 23 08:41:25.578 ERRO registering metric [deployment_eth_rpc_request_duration] because it was already registered, component: MetricsRegistry
Jun 23 08:41:25.579 ERRO registering metric [deployment_eth_rpc_errors] because it was already registered, component: MetricsRegistry
Jun 23 08:41:39.638 INFO Syncing 1 blocks from Ethereum, code: BlockIngestionStatus, blocks_needed: 1, blocks_behind: 1, latest_block_head: 10901392, current_block_head: 10901391, provider: rinkeby-rpc-0, component: BlockIngestor
Jun 23 08:41:40.810 INFO Done processing trigger, gas_used: 134710549, data_source: Metadata, handler: handleMetadataCreated, total_ms: 2, transaction: 0xe24d…1bb9, address: 0xfd8a…bff4, signature: MetadataCreated(indexed address,indexed address,bytes,bytes), block_hash: 0x10453a156b23f6b0cc0b40e1326b4e7f24c039d6d3e58a506be5d9a1f41cf193, block_number: 10901392, sgd: 3, subgraph_id: QmdNChb53fQdvKvndqT3dbM2Hcoo5rPqU45iHmnnNtinf4, component: SubgraphInstanceManager
Jun 23 08:41:40.817 INFO Applying 3 entity operation(s), block_hash: 0x10453a156b23f6b0cc0b40e1326b4e7f24c039d6d3e58a506be5d9a1f41cf193, block_number: 10901392, sgd: 3, subgraph_id: QmdNChb53fQdvKvndqT3dbM2Hcoo5rPqU45iHmnnNtinf4, component: SubgraphInstanceManager
Jun 23 08:41:54.985 INFO Syncing 1 blocks from Ethereum, code: BlockIngestionStatus, blocks_needed: 1, blocks_behind: 1, latest_block_head: 10901393, current_block_head: 10901392, provider: rinkeby-rpc-0, component: BlockIngestor
Jun 23 08:41:56.193 WARN ************************ handleNewToken: datatokenId 0x11aba8fb66ccfa232aa9a29fb533a8aa5a94192f, data_source: DTFactory, block_hash: 0x991edf3afbb9bf2ba49a17607c44a192319c27092a399989336356755669b2e8, block_number: 10901393, sgd: 3, subgraph_id: QmdNChb53fQdvKvndqT3dbM2Hcoo5rPqU45iHmnnNtinf4, component: SubgraphInstanceManager
Jun 23 08:41:56.195 INFO Create data source, params: 0x11aba8fb66ccfa232aa9a29fb533a8aa5a94192f, name: DataToken, block_hash: 0x991edf3afbb9bf2ba49a17607c44a192319c27092a399989336356755669b2e8, block_number: 10901393, sgd: 3, subgraph_id: QmdNChb53fQdvKvndqT3dbM2Hcoo5rPqU45iHmnnNtinf4, component: SubgraphInstanceManager
Jun 23 08:41:56.198 INFO Done processing trigger, gas_used: 365079003, data_source: DTFactory, handler: handleNewToken, total_ms: 1, transaction: 0xea6e…4b4a, address: 0x3fd7…e83a, signature: TokenRegistered(indexed address,string,string,uint256,indexed address,indexed string), block_hash: 0x991edf3afbb9bf2ba49a17607c44a192319c27092a399989336356755669b2e8, block_number: 10901393, sgd: 3, subgraph_id: QmdNChb53fQdvKvndqT3dbM2Hcoo5rPqU45iHmnnNtinf4, component: SubgraphInstanceManager
Jun 23 08:41:56.203 INFO Applying 3 entity operation(s), block_hash: 0x991edf3afbb9bf2ba49a17607c44a192319c27092a399989336356755669b2e8, block_number: 10901393, sgd: 3, subgraph_id: QmdNChb53fQdvKvndqT3dbM2Hcoo5rPqU45iHmnnNtinf4, component: SubgraphInstanceManager
Jun 23 08:41:56.240 ERRO registering metric [deployment_eth_rpc_request_duration] because it was already registered, component: MetricsRegistry
Jun 23 08:41:56.240 ERRO registering metric [deployment_eth_rpc_errors] because it was already registered, component: MetricsRegistry
```
